File: BRITED_STUDIO_FINAL/video_pipeline/ocr_frames.m
#import <Foundation/Foundation.h>
#import <Vision/Vision.h>

int main(int argc, const char * argv[]) {
  @autoreleasepool {
    NSMutableDictionary *result=[NSMutableDictionary dictionary];
    for(int i=1;i<argc;i++){
      NSString *path=[NSString stringWithUTF8String:argv[i]];
      NSURL *url=[NSURL fileURLWithPath:path];
      VNRecognizeTextRequest *request=[[VNRecognizeTextRequest alloc] init];
      request.recognitionLevel=VNRequestTextRecognitionLevelFast;
      request.usesLanguageCorrection=NO;
      VNImageRequestHandler *handler=[[VNImageRequestHandler alloc] initWithURL:url options:@{}];
      NSError *error=nil; [handler performRequests:@[request] error:&error];
      NSMutableArray *lines=[NSMutableArray array];
      if(!error) for(VNRecognizedTextObservation *obs in request.results){VNRecognizedText *text=[[obs topCandidates:1] firstObject];if(text) [lines addObject:text.string];}
      result[path]=[lines componentsJoinedByString:@" "];
    }
    NSData *json=[NSJSONSerialization dataWithJSONObject:result options:0 error:nil];
    fwrite(json.bytes,1,json.length,stdout);
  }
  return 0;
}

File: BRITED_STUDIO_FINAL/video_pipeline/ocr_frames.swift
import AppKit
import Foundation
import Vision

var result: [String: String] = [:]
for argument in CommandLine.arguments.dropFirst() {
    let url = URL(fileURLWithPath: argument)
    guard let image = NSImage(contentsOf: url),
          let data = image.tiffRepresentation,
          let bitmap = NSBitmapImageRep(data: data),
          let cgImage = bitmap.cgImage else {
        result[argument] = ""
        continue
    }
    let request = VNRecognizeTextRequest()
    request.recognitionLevel = .accurate
    request.recognitionLanguages = ["fr-FR"]
    request.usesLanguageCorrection = true
    try? VNImageRequestHandler(cgImage: cgImage).perform([request])
    let text = (request.results ?? []).compactMap {
        $0.topCandidates(1).first?.string
    }.joined(separator: " ")
    result[argument] = text
}
let output = try! JSONSerialization.data(withJSONObject: result)
FileHandle.standardOutput.write(output)
